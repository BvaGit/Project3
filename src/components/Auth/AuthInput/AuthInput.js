import React, { useState} from 'react';

import { validateRegField } from '/src/helpers/validation'

import Input from '../../Input';

const AuthInput = ({ value, changeField, id, label, placeholder, error, setError }) =>{
    const [blured, setBlured] = useState(false);

    const handleBlur = () =>{
        setBlured(true);
        setError({[id]: validateRegField(id, value) })
    }

    const handleChange = (e) => {
        changeField({ [id]: e.target.value })
        if(blured) setBlured(false)
    }

    return (
        <>
            <Input
                label={label}
                type="text"
                name={id}
                value={value}
                placeholder={placeholder}
                minLength="1"
                maxLength="30"
                onBlur={handleBlur}
                onChange={handleChange}
            />
            {(blured && error) && <div className="inputErr">{error}</div>}
        </>
    )
}

export default AuthInput 
