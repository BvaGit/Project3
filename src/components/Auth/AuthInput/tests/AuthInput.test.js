import React from 'react';
import { shallow, mount } from 'enzyme';
import AuthInput from '../AuthInput';
import mapStateToProps from '../index'
import { Provider } from 'react-redux';
import { IntlProvider } from "react-intl";


describe('AuthInput', () => {
    it("Should match snapshot", () => {
        const component = shallow(<AuthInput />)
        expect(component).toMatchSnapshot()
    })
    it("Should change input", () => {
        let props={rooms:{chat_id:1},socket:{emit:jest.fn()}, userId:1,changeField:jest.fn(), messages:{length:2}}
        const component = mountBiggerSmart(<AuthInput {...props}/>)
        component.find('input').simulate("change", {target:{value:"value"}})
        expect(component.find('input').value).toBe()
    })
})

describe('mapStateToProps', () =>{
    it('should be defined', function(){
        expect(mapStateToProps).toBeDefined()
    })
})

const mountBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return mount(<IntlProvider >{core}</IntlProvider>);
};