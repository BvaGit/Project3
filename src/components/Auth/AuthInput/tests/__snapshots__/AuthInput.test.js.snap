// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthInput Should match snapshot 1`] = `ShallowWrapper {}`;
